cmd/resultscollector/main.go

```diff
@@ -219,36 +253,22 @@ func main() {
 				fmt.Println(err)
 				os.Exit(1)
 			}
-			file := waitForResultsFile(scapresultsconf.File, scapresultsconf.Timeout)
-			defer file.Close()
 
-			contents, err := ioutil.ReadAll(file)
+			arfContents, err := readResultsFile(scapresultsconf.ArfFile, scapresultsconf.Timeout, scapresultsconf.Compress)
 			if err != nil {
 				fmt.Println(err)
 				os.Exit(1)
 			}
 
-			compressed := false
-			if resultNeedsCompression(contents) || scapresultsconf.Compress {
-				contents, err = compressResults(contents)
-				fmt.Println("Needs compression.")
-				if err != nil {
-					fmt.Println("Error: Compression failed")
-					os.Exit(1)
-				}
-				compressed = true
-				fmt.Printf("Compressed results are %d bytes in size\n", len(contents))
-			}
-
 			err = backoff.Retry(func() error {
 				url := scapresultsconf.ResultServerURI
 				fmt.Printf("Trying to upload to resultserver: %s\n", url)
-				reader := bytes.NewReader(contents)
+				reader := bytes.NewReader(arfContents.contents)
```

Comment:
why not use the Reader directly instead of reading the ARF and then creating a Reader out of the string?

Reply:
Because of the compression. I'm not sure there is a way to 'stream' the compression so that you would just have a reader that compresses what you read on the fly.
But if you think the interface would have been nicer if we had a reader returned (I do think it would be a bit nicer), then of course I can change the semantics to populate a reader in the `resultFileContents` struct. But, in case the file needs to be compressed, I don't think we'll get away with not reading the file.

Reply:
Right, if it needs to be compressed we do need to read the file, that's true. But that would still reduce the number of reads we have to do in total. Feel free to leave this as-is, and we can optimize this later.